util/reggen/multi_register.py

```diff
@@ -140,6 +147,14 @@ def get_field_list(self) -> List[Field]:
     def is_homogeneous(self) -> bool:
         return self.reg.is_homogeneous()
 
+    def handle_async(self, clocks: Clocking) -> object:
+        # multi reg must all have the same async clock
+        clock = None
+        for r in self.regs:
+            clock = r.handle_async(clocks)
```

Comment:
I agree with the comment. Maybe it would be worth adding an assertion to check it's true?

Reply:
good idea.